Separate `settings` view mode from `ContactInformation`

tomalec opened the pull request:
⚠ this PR requires https://github.com/woocommerce/google-listings-and-ads/pull/1025

### Changes proposed in this Pull Request:
- Separate `settings` view mode from `ContactInformation`
	into two "pages" components, for phone and address settings.
- Addapt StoreAddressCard's UI to the latest designs.

Implements part of https://github.com/woocommerce/google-listings-and-ads/issues/962#issuecomment-925686034.


### Screenshots:
[`/wp-admin/admin.php?page=wc-admin&path=%2Fgoogle%2Fsetup-mc`](https://gla1-test/wp-admin/admin.php?page=wc-admin&path=%2Fgoogle%2Fsetup-mc)
![image](https://user-images.githubusercontent.com/17435/135356099-9970c877-760b-4ed0-8981-0588d23e7a21.png)

[`/wp-admin/admin.php?page=wc-admin&subpath=%2Fedit-phone-number&path=%2Fgoogle%2Fsettings`](https://gla1.test/wp-admin/admin.php?page=wc-admin&subpath=%2Fedit-phone-number&path=%2Fgoogle%2Fsettings)
![image](https://user-images.githubusercontent.com/17435/135356805-197ac90c-e58f-4686-b73e-f0f1641048b6.png)


[`/wp-admin/admin.php?page=wc-admin&subpath=%2Fedit-store-address&path=%2Fgoogle%2Fsettings`](https://gla1.test/wp-admin/admin.php?page=wc-admin&subpath=%2Fedit-store-address&path=%2Fgoogle%2Fsettings)
![image](https://user-images.githubusercontent.com/17435/135356276-d4788f86-85ad-40f0-9b33-135501aba51c.png)


### Detailed test instructions:

#### Onboarding
1. Go to the [last step of MC setup](https://468b-193-239-83-51.ngrok.io/wp-admin/admin.php?page=wc-admin&path=%2Fgoogle%2Fsetup-mc)
2. See if the UI looks like in the latest Figma designs https://www.figma.com/file/eQ9O4m2flzrcAiDMXkOW0m/Google-Listings-%26-Ads-v1.x?node-id=1595%3A26419
3. Check if verifying a phone number, and saving the store address works as before
#### Settings
1. Go to [the GLA settings page](https://gla1.test/wp-admin/admin.php?page=wc-admin&path=%2Fgoogle%2Fsettings)
2. Click Phone number "Edit"
3. See if the UI looks like in the latest Figma designs 
3. Check if verifying a phone number works as before

1. Go to [the GLA settings page](https://gla1.test/wp-admin/admin.php?page=wc-admin&path=%2Fgoogle%2Fsettings)
2. Click Store address "Edit"
3. See if the UI looks like in the latest Figma designs 
3. Check if verifying saving the address works as before.

<!--
Optional.
Enter a summary of all changes in this Pull Request, which will be added to the changelog if accepted.
Each line should start with `(Fix|Add|Tweak|Update) - `, for example:
> Fix - I took care of something that wasn't working.
> Add - I added something new that's pretty cool.
> Tweak - I made a small change.
> Update - I made big changes to something that wasn't broken.

Leave the "Changelog entry" header in place completely empty, without any summary if no changelog entry is needed.
If you remove the "Changelog entry" header, the title of Pull Request will be used as the changelog entry.
-->
### Changelog entry

> Update phone number and store address pages flow.


### Additional notes:

- I changed the reload icon to Gutenbergs's "update" for better UI consistency, I hope that's fine.
- I changed the copy of Phone/Address section descriptions to a tiny bit less redundant:
	> Your ~contact information~**phone number** is required by Google for verification purposes. It will be shared with the Google Merchant Center and will not be displayed to customers.

tomalec committed:
Separate `settings` view mode from `ContactInformation`,

into two "pages" components, for phone and address settings.

Implements part of https://github.com/woocommerce/google-listings-and-ads/issues/962#issuecomment-925686034.

tomalec committed:
Addapt `StoreAddressCard`'s UI to the latest designs.

Implements part of https://github.com/woocommerce/google-listings-and-ads/issues/962#issuecomment-925686034

eason9487 commented on js/src/components/contact-information/index.js:
```diff
@@ -67,15 +67,13 @@ export function ContactInformationPreview() {
-export default function ContactInformation( { view, onPhoneNumberVerified } ) {
+export default function ContactInformation( { onPhoneNumberVerified } ) {
 	const phone = useGoogleMCPhoneNumber();
-	const isSetupMC = view === 'setup-mc';
```
💅 
The related code within the comment needs an update as well.

https://github.com/woocommerce/google-listings-and-ads/blob/4de904a98e44aa45d839fb86043975feba0aba6e/js/src/components/contact-information/index.js#L82-L84

tomalec replied:
Addressed in https://github.com/woocommerce/google-listings-and-ads/pull/1029/commits/920ebd6ec520387996c9685e2ec466ebe7e11c57

eason9487 commented on js/src/components/contact-information/store-address-card.scss:
```diff
@@ -0,0 +1,22 @@
+.gla-store-address-card {
+	// We could've used `StoreAddressCard'`s selectors,
+	// but the elements are wrapped with `<FlexItem>`,
+	// which does not have accessible semantic selector.
+	// So we have to rely on `<AccountCard>`'s internal implementation details.
```
💅 

What about adding `className` onto these components?

https://github.com/woocommerce/google-listings-and-ads/blob/4de904a98e44aa45d839fb86043975feba0aba6e/js/src/components/account-card/index.js#L78

https://github.com/woocommerce/google-listings-and-ads/blob/4de904a98e44aa45d839fb86043975feba0aba6e/js/src/components/account-card/index.js#L84

tomalec replied:
I proposed that with a bit more structured & generic way in https://github.com/woocommerce/google-listings-and-ads/pull/1030

eason9487 reviewed (approved):
Tested locally and works well. LGTM.

tomalec committed:
Update a code comment to the changed logic.

Addresses https://github.com/woocommerce/google-listings-and-ads/pull/1029#discussion_r719061183.
